File: CarAuction.Backend/CarAuction.Domain/Entities/Group.cs
namespace CarAuction.Domain.Entities;

public class Group
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int? CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ICollection<GroupMember> Members { get; set; } = new List<GroupMember>();
    public ICollection<GroupMessage> Messages { get; set; } = new List<GroupMessage>();
}

File: CarAuction.Frontend/src/types/group.ts
export interface Group {
  id: number;
  name: string;
  description: string;
  createdBy: number;
  createdAt: string;
  members?: GroupMember[];
}

export interface GroupMember {
  userId: number;
  firstName: string;
  lastName: string;
  joinedAt: string;
}

export interface CreateGroupRequest {
  name: string;
  description: string;
}

export interface GroupMessage {
  id: number;
  groupId: number;
  senderId: number;
  senderName?: string;
  content: string;
  sentAt: string;
}

export interface GroupMessageDto {
  id: number;
  groupId: number;
  senderId: number;
  senderName: string;
  content: string;
  sentAt: string;
}
